src/nccl_ofi_rdma.c

```diff
@@ -2955,7 +2955,16 @@ static int reg_mr_ep(nccl_net_ofi_rdma_ep_t *ep,
 		/* Cache miss */
 	}
 
-	ret = reg_mr_on_device(ep, ckey, type, &ret_handle);
+	if (mr_cache && nccl_ofi_mr_ckey_is_iovec(ckey)) {
+		/*
+		 * Use page aligned address to register for smaller allocations.
+		 * Not needed for DMA buffers.
```

Comment:
nit: i'd prefer "dmabuf" here for clarity. All dmabufs are DMA buffers, but not all DMA buffers are dmabufs.

Reply:
also, this comment makes no sense.  You still need the cache for dma-buf, so the underlying problem remains and this special case shouldn't be here.